### PR TITLE
feat: change default precisions and make them configurable

### LINKED ISSUE #16: Standard data types and data length
Hi
I have been busy with other stuff so it took me some time to respond to @argoyle call for tests. and hope that @henricm agrees that it is time to fix this issue.
I have tested new version and it seems to work but there is some issues that was blocking me from using it and that is related to data types standard because I need more accurate numbers in my FA integration in HA

The old version of the code is now blocked after core-2021.5.5 due to missing key and not following the HA code standard so it is time to do something about it and summer time and more to off work coding :)

I have been using the data types below and want you align with the ide so I wont make a PR of is as of now (you can do it if you agree with the change, I have added a link to a updated script below. 

The data standard that I recommend is listed below
Integer
W - same as before
°C - No need for decimals to fill up the database. (2 dec before)
V - same as before
VA - same as before
Wh - same as before

Float with one decimal 
% - Its just not enough accurate  (INT before)
kWh - Logging energy usage per dag (INT before)
A - No need for decimals to fill up the database. (2 dec before)
kW - No need for decimals to fill up the database. (2 dec before)

I made a version of the sensor.py today that fixed the issues above and fixed som naming standard stuff etc. 
https://github.com/danielolsson100/ha-ferroamp/blob/danielolsson100-patch-1/custom_components/ferroamp/sensor.py

output example
image
image

### PR DESCRIPTION
**Pull request by argoyle**
It's now possible to set the precision for battery, current, energy, power, temperature and voltage in the option flow.

Closes #16

### CODE REVIEW

**Comment by argoyle** [comment]
Not sure about the default precisions so please have opinions. 😄

File: custom_components/ferroamp/config_flow.py [hunk 1]
```diff
@@ -56,13 +66,61 @@ async def async_step_init(self, user_input=None):
...
+        if precision_current is None:
+            precision_current = 0
+
+        precision_energy = self.config_entry.options.get(CONF_PRECISION_ENERGY)
+        if precision_energy is None:
+            precision_energy = 1
+
+        precision_power = self.config_entry.options.get(CONF_PRECISION_POWER)
+        if precision_power is None:
+            precision_power = 0
```
Comment (henricm) [inline custom_components/ferroamp/config_flow.py:83]: This (precision=0) will result in power for SSO which now is in kW will only display 1 kW if eg 1.34 kW, right? I think that's too inexact as default for raw data like this. Default should be 2 if we want to keep it as kW.

**Review by henricm — CHANGES_REQUESTED** [review]
Currently SSO and ESO power is calculated from Voltage and Current and then converted to kW. I think it would be better to change all Power sensors to W (Watt) instead of kW. And leave it up to the user/presentation layer to convert it to kW, MW etc if wanted. If we do that, there should be no need for precision on Power.

What do you think? I understand it will mess up historical comparisons, but this code is also very much still under development so it might be worth fixing this as early as possible.

File: custom_components/ferroamp/config_flow.py (hunk 1 above)
Comment (argoyle) [reply custom_components/ferroamp/config_flow.py:83]: True. Changing. Agree on leaving formatting to presentation layer.

Comment (argoyle) [reply custom_components/ferroamp/config_flow.py:83]: Hmmm, I use the power precision in the `ThreePhasePowerFerroampSensor` and there are decimals there. Not sure they are needed ever though. Thoughts @henricm ?

Comment (argoyle) [reply custom_components/ferroamp/config_flow.py]: Removed power precision completely

**Review by henricm — APPROVED** [review]
Looks great! Good work.

**Pull request merged**